### PR TITLE
Use internal ACLs for new flaws

### PR DESCRIPTION
**Pull request by superbuggy**
There are still considerable logic that uses the PUBLIC groups, and conceivably those will need to be adapted. Is it necessary to change to use `settings.INTERNAL_`{READ/WRITE}`_GROUP` everywhere? It seems like it would but it certainly seems to enlarge the impact and scope of this change. Should I abort this approach to preventing the emails? I wonder if there is a more granular, well-scoped approach.

Closes OSIDB-4274

### CODE REVIEW

**Review by osoukup — CHANGES_REQUESTED** [review]
I stopped the review after spotting a few similar issues as I think there is some misunderstanding. This task is not about changing how embargo works. This task is about changing the ACLs on manual flaw creation from public to internal. No other behaviour should be changed.

File: osidb/serializer.py [hunk 1]
```diff
@@ -429,10 +429,10 @@ def get_acls(self, embargoed):
         generate ACLs based on embargo status
         """
         acl_read = (
-            settings.EMBARGO_READ_GROUP if embargoed else settings.PUBLIC_READ_GROUPS
+            settings.EMBARGO_READ_GROUP if embargoed else settings.INTERNAL_READ_GROUP
         )
         acl_write = (
-            settings.EMBARGO_WRITE_GROUP if embargoed else settings.PUBLIC_WRITE_GROUP
+            settings.EMBARGO_WRITE_GROUP if embargoed else settings.INTERNAL_WRITE_GROUP
```
[the remark this answers is not in the document]
Comment (superbuggy) [reply osidb/serializer.py:435]: That makes more sense... would I want to hardcode this `ACLMixin` then (feels wrong to do that in `Flaw::save`)?
```py
   def save(self, *args, **kwargs):
        if not self.acl_read and not self.acl_write:
            from django.conf import settings
            from osidb.core import generate_acls

            self.acl_read = [
                uuid.UUID(acl) for acl in generate_acls([settings.INTERNAL_READ_GROUP])
            ]
            self.acl_write = [
                uuid.UUID(acl) for acl in generate_acls([settings.INTERNAL_WRITE_GROUP])
            ]
```

This feels kind of

[the remark this answers is not in the document]
Comment (superbuggy) [reply osidb/serializer.py:435]: Thank you for your helpful thoughts on this PR, @osoukup!

File: osidb/mixins.py [hunk 2]
```diff
@@ -206,6 +206,17 @@ def __init__(self, *args, **kwargs):
         # caching the ACLs and building the group map
         self.acls_all
 
+    def save(self, *args, **kwargs):
+        if not self.acl_read and not self.acl_write:
```
Comment (osoukup) [inline osidb/mixins.py:210]: I think this assumes that there are no ACLs set when a user creates the flaw via API. **embargoed** is however a mandatory flaw parameter and it is always translated into either public or embargo ACLs.

**Review by osoukup — CHANGES_REQUESTED** [review]
This solution does not seem correct to me. Anyway, I strongly recommend to create an integration test which will verify that a user-created flaw has internal visibility as intended.

File: osidb/tests/test_e2e.py [hunk 3]
```diff
@@ -644,3 +645,188 @@ def test_cveorg_with_workflows(
...
+        updated_flaw_data = {
+            "title": "Updated Test ACL values",
+            "comment_zero": "This is an updated test for ACL validation",
+            "impact": "IMPORTANT",
+            "components": ["curl"],
+            "source": "REDHAT",
+            "reported_dt": "2024-08-06T00:00:00.000Z",
+            "unembargo_dt": "3339-08-06T00:00:00.000Z",
+            "updated_dt": flaw.updated_dt,
+            "embargoed": True,  # Change to embargoed
```
Comment (osoukup) [inline osidb/tests/test_e2e.py:788]: Changing the visibility to something more strict does not make much sense and ideally/hopefully would be prevented by the validation. The reason is that when something was already visible publicly or internally, it is can be already cached somewhere and trying to hide it is already futile.

**Review by osoukup — CHANGES_REQUESTED** [review]
I have a few comments, I do not want to look to pedantic but I think we still need to polish this. The test is a great addition!

File: osidb/tests/test_e2e.py (hunk 3 above)
Comment (superbuggy) [reply osidb/tests/test_e2e.py:788]: I wondered if it didn't make sense, but then wondered if there are ever edge cases where something needs to become embargoed, but upon looking at it again, a user wouldn't send this type of payload (`"embargoed": True`) with a PUT request. Should I strengthen the validation? This didn't trigger any complaint

File: osidb/serializer.py [hunk 4]
```diff
@@ -428,11 +428,20 @@ def get_acls(self, embargoed):
         """
         generate ACLs based on embargo status
         """
+        internal = self.context["request"].method == "POST"
```
[the remark this answers is not in the document]
Comment (superbuggy) [reply osidb/serializer.py:431]: My first impulse was to check the pk/uuid (in def `embargo2acls`), but wasn't sure how to reliably access that data so I ended up with this much less felicitous approach here. I was trying stuff like `self.Meta.model.instance` but not seeing what I was expecting to.

**Comment by superbuggy** [comment]
Thank you for thoughtful thoroughness here, @osoukup

File: osidb/tests/test_e2e.py (hunk 3 above)
Comment (osoukup) [reply osidb/tests/test_e2e.py:788]: So there actually is that validation https://github.com/RedHatProductSecurity/osidb/blob/master/osidb/models/flaw/flaw.py#L691 but again only meant for the binary embargo/public world so does not get triggered by internal-to-embargoed change. You may generalize it so it works for any late attempt to restrict the access.
